examples/litepeer/litepeer.go

```diff
@@ -77,38 +78,36 @@ func main() {
 	if len(*sharable) == 0 {
 		returnError("Sharable string not provided", true)
 	}
-	lc, err := lib.NewLightClient(*destination, *timeout)
+	lc, err := lib.NewLightClient(*destination, *timeout, *jsonOut)
 	if err != nil {
 		returnError("Failed setting up client reason:"+err.Error(), true)
 	}
 	var upd lib.ProgressUpdater
 	upd = &noopProgress{}
 	if !*onlyInfo && *showProg {
-		upd = &updateProgress{}
+		upd = &updateProgress{
+			jsonOut: *jsonOut,
+		}
 	}
-	status, err := lc.Start(*sharable, *onlyInfo, *stat, upd)
-	if err != nil {
-		returnError(status+" reason:"+err.Error(), false)
-	}
-	fmt.Println(status)
+	out := lc.Start(*sharable, *onlyInfo, *stat, upd)
+	lib.OutMessage(out, *jsonOut)
 	return
 }
 
 type noopProgress struct{}
 
-func (u *noopProgress) UpdateProgress(p int) {
+func (u *noopProgress) UpdateProgress(p, downloadedSize, fullSize int) {
 	return
 }
 
 type updateProgress struct {
 	started bool
+	jsonOut bool
 }
 
-func (u *updateProgress) UpdateProgress(p int) {
-	if u.started {
-		CallClear()
-	} else {
-		u.started = true
-	}
-	fmt.Printf("Progress %d%%\n", p)
+func (u *updateProgress) UpdateProgress(p, downloadedSize, fullSize int) {
+	dMB := float32(downloadedSize) / (1024 * 1024)
+	fMB := float32(fullSize) / (1024 * 1024)
+	out := lib.NewOut(200, "Progress", "", fmt.Sprintf("%d%% (%.2fMB / %.2fMB)", p, dMB, fMB))
```

Comment:
Everything is INT here... why are we using 0.2f format specifier?

Reply:
```
dMB := float32(downloadedSize) / (1024 * 1024)
	fMB := float32(fullSize) / (1024 * 1024)
```

using `%.2f` after converting bytes into mb